cms/static/cms/js/modules/cms.toolbar.js

```diff
@@ -521,7 +521,16 @@ var CMS = window.CMS || {};
                 var text = opts.text || '';
                 var callback = opts.callback;
                 var onSuccess = opts.onSuccess;
-                var question = (text) ? confirm(text) : true;
+                // a user could click on "Prevent this page form creating additional
+                // dialogs." which would prevent the modal to be closed
+                // We fallback to "true" in this case to avoid this
+                var secureConfirm = function (message) {
+                    var start = Number(new Date());
+                    var result = confirm(message);
+                    var end = Number(new Date());
+                    return (end < (start + 10) || result === true);
```

Comment:
if confirm dialog is prevented - then it is false, but first condition is true - so it defaults to true.
if confirm dialog is _not_ prevented then it is false (and we need to just return it)